src/index.js

```diff
@@ -56,29 +62,34 @@ const elmWebComponents = {
 
     class ElmElement extends HTMLElement {
       connectedCallback() {
-        let props = Object.assign({}, getProps(this), staticFlags)
-        if (Object.keys(props).length === 0) props = undefined
-
-        const flags = mapFlags(props)
-
-        if (elmVersion === '0.19') {
-          /* a change in Elm 0.19 means that ElmComponent.init now replaces the node you give it
-           * whereas in 0.18 it rendered into it. To avoid Elm therefore destroying our custom element
-           * we create a div that we let Elm render into, and manually clear any pre-rendered contents.
-           */
-          const elmDiv = document.createElement('div')
-
-          this.innerHTML = ''
-          this.appendChild(elmDiv)
-
-          const elmElement = ElmComponent.init({
-            flags,
-            node: elmDiv,
-          })
-          setupPorts(elmElement.ports)
-        } else if (elmVersion === '0.18') {
-          const elmElement = ElmComponent.embed(this, flags)
-          setupPorts(elmElement.ports)
+        try {
+          let props = Object.assign({}, getProps(this), staticFlags)
+          if (Object.keys(props).length === 0) props = undefined
+
+          const flags = mapFlags(props)
+
+          if (elmVersion === '0.19') {
+            /* a change in Elm 0.19 means that ElmComponent.init now replaces the node you give it
+             * whereas in 0.18 it rendered into it. To avoid Elm therefore destroying our custom element
+             * we create a div that we let Elm render into, and manually clear any pre-rendered contents.
+             */
+            const elmDiv = document.createElement('div')
+
+            this.innerHTML = ''
+            this.appendChild(elmDiv)
+
+            const elmElement = ElmComponent.init({
+              flags,
+              node: elmDiv,
+            })
+            setupPorts(elmElement.ports)
+          } else if (elmVersion === '0.18') {
+            const elmElement = ElmComponent.embed(this, flags)
+            setupPorts(elmElement.ports)
+          }
+        } catch (error) {
+          onSetupError(error)
+          throw error
```

Comment:
I think we should only throw the error if the user hasn't passed an `onSetupError`. If I'd passed an `onSetupError` I think I'd be surprised to see the error also being thrown by the library. What do you think?

Reply:
I understand your point. I added the `throw` so that during development something still gets logged to console since re-throwing the error shouldn't affect the application flow. Maybe using `console.error` would make the intention more clear. I will try `console.error` in development and push something to production to see if it behaves as expected.